ipynb/wltests/sched-evaluation-full.ipynb

```diff
@@ -130,7 +130,7 @@
     "for test in collector.tests(workload='jankbench'):\n",
     "    logging.info(\"Results for: %s\", test)\n",
     "    collector.report(workload='jankbench', metric='device_total_energy',\n",
-    "                     test=test, sort_on='99%', ascending=True)"
+    "                     test=\"^{}$\".format(test), sort_on='99%', ascending=True)"
```

Comment:
I think all energy plots should be sorted by "mean" instead of 99% percentile. Could you quickly fix that at the same time ?

Reply:
I would say that the 99th percentile is more restrictive and thus is a better default.
Users can always look at the mean by changing this value.... but at least by default we are much more confident about a 99th percentile improvement than an average... especially if the std is relatively high.

Thus, I would propose to keep it as that by default.

Reply:
I don't understand what you mean by "more restrictive" ? It's just a different comparison criteria for me. I understand why the 99% percentile makes sense for dropped frames but, for energy, the average really is what we care about.

Reply:
Here we are boxplotting `device_total_energy`, which is already a kind of "average"  of all the samples for one experiment. Thus, let say you have only two iterations and these two different kernels with the corresponding `device_total_energy` for each iteration:
```
Kernel A)  device_total_energy = [ 100W, 200W ]  => avg: 150W
Kernel B)  device_total_energy = [ 130W, 180W ]  => avg: 155W
```
Which one do we prefer?

Reply:
`device_total_energy` is just the sum of the energy on all channels no ? It should be in Joules, not Watts. I'm not sure to understand when you say it's a kind of average ...

Reply:
And I think talking about the 99% percentile vs average on just two samples isn't fair. If you have
```
Kernel A) device_total_energy = [ 100J, 100J, 100J, 100J, 200J] => avg: 133J
Kernel B) device_total_energy = [ 180J, 180J, 180J, 180J, 180J] => avg: 180J
```
Which one is the best ?

Reply:
Two samples was (of course) an intentionally simplified example ;-)

Regarding your example with 5 sample, I would choose the first kernel of course, but at the same time I would really wanna know why kernel A generated such a bad execution.
Thus, still I would prefer to have a first sanity check on 99 percentile.

At the end, IMO these plots are not for choosing in an automated way one over another, but instead to highlight potential issues to be better investigated.

Reply:
Well, these plots are here to compare kernels, and we're talking about a sorting criteria so this is about telling which one is better than another. Is the average energy the only criteria to take into account ? Definitely not, but it's clearly the one we usually care about.

When I run this notebook I always start by changing the sorting criteria for energy plots to "mean" simply because 99% of the time I don't really care about the the 99% percentile. If I see one suspicious kernel with bad p-values or something like that then I'll look into more details but by default I'd like my plots to be sorted on something that is usually the most sensible option.

Reply:
The point is actually to define what "is usually the most sensible option" and not for your specific need but for a generic case. And to me that's the 99 percentile, for energy as well as for total frame duration.

Average by itself is not reporting sensible information, at least when not paired with stddev.
Thus, if I have to order based on a single metric I prefer the 95 or 99 percentile.

Now, we run 30 iterations of each Jankbench test and we collect one sample of overall energy consumption for each iteration.  Using an numerical example where just one 1 sample reports 5% more energy:

```python
data = [100 for _ in range(29)] + [105]
df = pd.DataFrame(data)
stats = df.describe(percentiles=[.75, .95, .99]).T
print stats
   count        mean       std    min    50%    75%    95%     99%    max
0   30.0  100.166667  0.912871  100.0  100.0  100.0  100.0  103.55  105.0
```
The 99 percentile will "classify" this as a 3.5% overhead (wrt to 100J mean score).
Which is just slightly over the expected value considering the `3*stddev` rule:

```python
_max = stats['mean'].values[0] + 3 * stats['std'].values[0]
_max
102.9052794541925
```
which will cut at a 2.9% overhead... and that's just because we use a still "relatively" small number of sample for a normally distributed data.

Consider that also with a `2*stddev` margin:

```python
_max = stats['mean'].values[0] + 2 * stats['std'].values[0]
_max
101.99240852501723
```
we still get a ~2% overhead on energy consumption.

Now, you can argue that the single run with a measured 5% energy overhead is just an outlier, but that's a bit of a subjective definition, unless we wanna apply some more elaborate statistical analysis focused on outliers detection. Otherwise, a first analysis is already provided by the boxplot, if that point happen to be outside of the box, you can easily consider that value as an outlier and "promote" your result.

In general I would prefer that a 5% overhead triggers a bell and one should better have a good explanation for the reason of such a bad result.
It can sound like a "false negative is good" policy, but given the context in which we operate I would rather prefer to spent a bit more time to convince people about an outlier then merging something which really risks to burn battery more then expected even if only under certain specific conditions.
Of course all these considerations have to be adapted to the specific kind of proposed patch, e.g. I would accept an higher risk for a refactoring/simplifying patch then for one that adds a new feature.

However, since it seems that me and we cannot reach an agreement on that, let's just collect some more feedback and I'm ok to change it if most of the other users are find the 99 percentile too much pessimistic. In that case I would however suggest to come up with at least a `2*stdev` (i.e. 95 percentile) based ordering.

Reply:
I'm not arguing the 99% percentile isn't useful. You described a nice example and I can't agree more that all stats value produced with this tool can be useful. I'm saying the goal of this plots is to _compare_ kernels and when it comes to energy the first thing you'd like to know is whether the new feature you just implemented saved energy or not. So the question is: how do we define what 'saving energy' means ?
Again, is it enough to use just the average ? Definitely not. Indeed, an average without the associated stddev is poor. But one nice thing about these plots is that they show you how spread the data is, so you can have the feeling of both the average and  the stddev at once.

Anyway, I'm OK to collect feedback from other users. In the meantime, your fixes are more than welcome for this notebook so I'm happy to merge it as-is. I'll submit a PR to change from 99% to mean and we can go on with this discussion there ;)